feat: Copy to clipboard button for settings/runs in desktop GUI

kevcodez opened the pull request:
Adds the copy to clipboard function for the Project Id JSON and the Record Key

Closes #9002

### User facing changelog
* Within the desktop GUI, in the settings tab, the user can now quickly copy the project id configuration and record key using a copy to clipboard icon.
* Within the desktop GUI, in the run tab, if a user no runs yet, the user can quickly copy project id configuration and record command using a copy to clipboard icon.

### Additional details
This change improves accessibility. A user no longer needs to manually select and copy the text.

### How has the user experience changed?
* Under the Settings tab in the desktop GUI, the code fields for the project id configuration and the record command (under Reord Key) have a copy to clipboard icon.
* Under the Run tab, when the connection to the dashboard is set up, but there has been no runs yet, new copy to clipboard icons allow easy copying of configuration and command

<img width="442" alt="Screenshot 2020-10-29 at 16 13 27" src="https://user-images.githubusercontent.com/14073399/97593518-0de06a80-1a02-11eb-9666-0569caf1f0ac.png">
<img width="594" alt="Screenshot 2020-10-29 at 16 14 40" src="https://user-images.githubusercontent.com/14073399/97593534-1173f180-1a02-11eb-87ef-d4f83494819a.png">
<img width="596" alt="Screenshot 2020-10-29 at 16 15 00" src="https://user-images.githubusercontent.com/14073399/97593537-120c8800-1a02-11eb-9548-f23dc9e4ea10.png">


### PR Tasks
* [x] Have tests been added/updated?
* [ ] Has the original issue or this PR been tagged with a release in ZenHub? <!-- (internal team only)-->
* [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
* [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
* [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cypress-io/cypress?pullRequest=9003) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cypress-io/cypress?pullRequest=9003) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cypress-io/cypress?pullRequest=9003) it.</sub>

jennifer-shehane reviewed (changes requested):
I do see this is working as explained. 

![](http://g.recordit.co/q0JFiWLQwx.gif)

1. We'll need some tests around this functionality to ensure it's working in future versions.

	- This test would be located in https://github.com/cypress-io/cypress/blob/develop/packages/desktop-gui/cypress/integration/settings_spec.js#L100:L100
	- Example test for testing clipboard: https://github.com/cypress-io/cypress/blob/develop/packages/desktop-gui/cypress/integration/update_modal_spec.js#L115:L115

	There is a Contributing guide that covers how to contribute and get Cypress running locally in generally here: https://github.com/cypress-io/cypress/blob/develop/.github/CONTRIBUTING.md

	To run the tests:

	- within `cypress` root, run `yarn`
	- run `yarn workspace @packages/desktop-gui cypress:open`
	- click on the test you're writing to run it within Cypress

	Instructions for running the `desktop-gui` tests can always be found here: https://github.com/cypress-io/cypress/blob/develop/packages/desktop-gui/README.md

2. You may want to also add this functionality to our 'Runs' tab after a new project is setup, since it also shows the same view with the projectId and record key:

	<img width="760" alt="Screen Shot 2020-10-29 at 5 28 39 PM" src="https://user-images.githubusercontent.com/1271364/97560332-e8a52980-1a0c-11eb-8b28-fe0834799421.png">

kevcodez commented:
@jennifer-shehane I've added the functionality to the runs tab, added tests for all copy icons and used JSON.stringify for formatting the JSON rather than doing it via string, let me know if that looks fine to you

jennifer-shehane reviewed (approved):
The code looks fine and tested - will leave the rest of review to @CypressAdam. The only thing I can think to make it better is to have a tooltip on hover saying 'Copy to clipboard'.

jennifer-shehane reviewed (changes requested):
@kevcodez Actually there are some failing tests. Did you run the tests locally to make sure they pass?

<img width="951" alt="Screen Shot 2020-11-02 at 10 10 08 AM" src="https://user-images.githubusercontent.com/1271364/97827791-abce8080-1cf3-11eb-8e5a-05b0aed9ec59.png">

kevcodez commented:
I did, will check again though. Will also add a tooltip.

kevcodez commented:
@jennifer-shehane Added tooltips and fixed the tests (used arrow function by accident so `this` was binded differently).

It's a bit tough to check the tooltips using the tests. In the run list tests they looked fine, on the project id configuration the tooltip seemed to be a bit out of place. That might also just be an issue with the DOM snapshot - could you check if the tooltips look fine?

CypressAdam reviewed (commented):
@jennifer-shehane this functionally, visually lgtm! I think we introduced a similar pattern on the install dialog. Nice buff here @kevcodez

jennifer-shehane reviewed (approved):
You have to use `.trigger('mousover')` and `.trigger('mouseout')` to simulate hovering for the tooltip. I went ahead and added tests for this since I was already in the repo testing this out. 

Thanks again for the contribution!
